deltacat/tests/aws/test_s3u.py

```diff
@@ -10,3 +28,105 @@ def test_uuid_block_write_provider_sanity(self):
         result = provider("base_path")
 
         self.assertRegex(result, r"^base_path/[\w-]{36}$")
+
+
+@pytest.fixture(autouse=True, scope="module")
+def mock_aws_credential():
+    os.environ["AWS_ACCESS_KEY_ID"] = "testing"
+    os.environ["AWS_SECRET_ACCESS_ID"] = "testing"
+    os.environ["AWS_SECURITY_TOKEN"] = "testing"
+    os.environ["AWS_SESSION_TOKEN"] = "testing"
+    os.environ["AWS_DEFAULT_REGION"] = "us-east-1"
+    yield
+
+
+@pytest.fixture(scope="module")
+def setup_s3_resource(mock_aws_credential):
+    with mock_s3():
+        yield boto3.resource("s3")
+
+
+@pytest.fixture(autouse=True, scope="module")
+def setup_test_s3_bucket(setup_s3_resource: ServiceResource):
+    setup_s3_resource.create_bucket(
+        ACL="authenticated-read",
+        Bucket=TEST_S3_BUCKET_NAME,
+    )
+    yield
+
+
+def test_sanity(setup_test_s3_bucket):
```

Comment:
Can you create a separate test class for unit under test for modularity? This spaghetti test suite will soon become unmaintainable otherwise.

Reply:
+1